core/cas-server-core-util/src/main/java/org/apereo/cas/util/RegexUtils.java

```diff
@@ -15,7 +11,8 @@
  * @since 5.0.0
  */
 public final class RegexUtils {
-    private static final Logger LOGGER = LoggerFactory.getLogger(RegexUtils.class);
+
+    private static final Pattern MATCH_NOTHING_PATTERN = Pattern.compile("a^");
```

Comment:
This is not a pattern technically that matches nothing. It does match something!